timesketch/frontend-ng/src/components/Explore/AddManualEvent.vue

```diff
@@ -0,0 +1,182 @@
+            outlined
+            hide-details
+            v-on:click="showPicker = true"
+            v-model="datetime"
+          >
+          </v-text-field>
+        </v-col>
+      </v-row>
+
+      <v-row v-if="showPicker">
+        <v-col cols="12">
+          <date-picker
+            v-model="dateFromPicker"
+            mode="dateTime"
+            ref="picker"
+            timezone="UTC"
+            :is-dark="$vuetify.theme.dark"
+            is24hr
+            is-date
+            is-expanded
+          ></date-picker>
+        </v-col>
+      </v-row>
+
+      <v-row>
+        <v-col cols="6">
+          <v-text-field :value="message" v-model="message" label="Message" outlined hide-details> </v-text-field>
+        </v-col>
+      </v-row>
+
+      <v-row>
+        <v-col cols="6">
+          <v-text-field
+            :value="timestampDesc"
+            label="Timestamp Description"
+            v-model="timestampDesc"
+            outlined
+            hide-details
+          >
+          </v-text-field>
+        </v-col>
+      </v-row>
+
+      <v-row v-for="(attribute, index) in attributes" :key="index">
+        <v-col cols="6">
+          <v-text-field
+            label="Attribute Name"
+            outlined
+            hide-details
+            v-model="attributes[index].name"
+            @change="setAddAttributeBtn(index)"
+          >
+          </v-text-field>
+        </v-col>
+        <v-col cols="6">
+          <v-text-field
+            label="Attribute Value"
+            outlined
+            hide-details
+            v-model="attributes[index].value"
+            @change="setAddAttributeBtn(index)"
+          >
+          </v-text-field>
+        </v-col>
+      </v-row>
+
+      <v-card-actions>
+        <v-spacer></v-spacer>
+        <v-btn text color="primary" @click="addAttribute" :disabled="isDisabled"> Add Attribute </v-btn>
+        <v-btn text color="primary" @click="clearAndCancel"> Cancel </v-btn>
+        <v-btn text color="primary" @click="submit"> Add Event </v-btn>
+      </v-card-actions>
+    </v-container>
+  </v-card>
+</template>
+
+<script>
+import dayjs from '@/plugins/dayjs'
+import DatePicker from 'v-calendar/lib/components/date-picker.umd'
+import ApiClient from '../../utils/RestApiClient'
+
+export default {
+  props: ['datetimeProp'],
+  components: {
+    DatePicker,
+  },
+  data() {
+    return {
+      message: null,
+      timestampDesc: null,
+      filterTab: null,
+      showPicker: false,
+      attributes: [],
+      isDisabled: false,
+      datetime: this.datetimeProp,
+    }
+  },
+  watch: {
+    datetimeProp: function () {
+      this.datetime = this.datetimeProp
+    },
+  },
+  computed: {
+    dateFromPicker: {
+      set(val) {
+        if (val !== null) this.datetime = dayjs.utc(val).millisecond(0).toISOString()
+      },
+      get() {
+        return this.datetime
+      },
+    },
+  },
+  methods: {
+    setAddAttributeBtn: function (i) {
```

Comment:
As a second though, if the sole purpose of this function is to determine whether the button should be enabled (through the isDisabled attribute), I think a better way to reflect this would be to make isDisabled a computed property that does the check you do on L139 and returns true or false.

The only caveat that I can think of is whether the computed property will register changes on the attributes in objects contained in your list, but according to [this](https://vuejs.org/guide/essentials/reactivity-fundamentals.html#deep-reactivity) it should.

Reply:
Make sense!
What about something like this:
```
isDisabled() {
      return this.attributes.some((attribute) => {
        return attribute.name === '' || attribute.value === ''
      })
    },
```
In this way, also the previous input text will contribute to the final value of isDisabled